rpcs3/Emu/Cell/SPUASMJITRecompiler.cpp

```diff
@@ -1607,36 +1489,16 @@ void spu_recompiler::RDCH(spu_opcode_t op)
 	case SPU_RdEventMask:
 	{
 		const XmmLink& vr = XmmAlloc();
-		c->movd(vr, SPU_OFF_32(ch_event_mask));
+		c->movq(vr, SPU_OFF_64(ch_events));
+		c->psrldq(vr, 4);
 		c->pslldq(vr, 12);
 		c->movdqa(SPU_OFF_128(gpr, op.rt), vr);
 		return;
 	}
 	case SPU_RdEventStat:
 	{
 		spu_log.warning("[0x%x] RDCH: RdEventStat", m_pos);
-		get_events();
-		Label wait = c->newLabel();
-		Label ret = c->newLabel();
-		c->jz(wait);
-
-		after.emplace_back([=, this, pos = m_pos]
-		{
-			c->bind(wait);
-			c->lea(addr->r64(), get_pc(pos));
-			c->and_(*addr, 0x3fffc);
-			c->mov(SPU_OFF_32(pc), *addr);
-			c->mov(arg1->r32(), op.ra);
-			c->mov(*arg0, *cpu);
-			c->call(imm_ptr(spu_rdch));
-			c->jmp(ret);
-		});
-
-		c->bind(ret);
-		c->movd(x86::xmm0, *addr);
-		c->pslldq(x86::xmm0, 12);
-		c->movdqa(SPU_OFF_128(gpr, op.rt), x86::xmm0);
-		return;
+		break; // TODO
```

Comment:
Multiple TODOs with no real fallback provided here and in WRCH. Doesn't this trash asmjit? What is the plan on this?

Reply:
After the switch there is the main WRCH fallback, unless im missing something.

Reply:
The new event stuff also added the missing immediate interrupt checks which would be ugly to implement it ASMJIT.